packages/core/src/button/Button.tsx

```diff
@@ -36,32 +28,26 @@ export interface ButtonBaseProps<T extends ElementType> {
    * If `true`, the button will be focusable when disabled.
    */
   focusableWhenDisabled?: boolean;
-  onBlur?: (evt: FocusEvent<inferElementType<T>>) => void;
-  onClick?: (evt: MouseEvent<inferElementType<T>>) => void;
-  onKeyDown?: (evt: KeyboardEvent<inferElementType<T>>) => void;
-  onKeyUp?: (evt: KeyboardEvent<inferElementType<T>>) => void;
+
+  onBlur?: (evt: FocusEvent<HTMLButtonElement>) => void;
+  onClick?: (evt: MouseEvent<HTMLButtonElement>) => void;
+  onKeyDown?: (evt: KeyboardEvent<HTMLButtonElement>) => void;
+  onKeyUp?: (evt: KeyboardEvent<HTMLButtonElement>) => void;
   /**
    * The variant to use. Options are 'primary', 'secondary' and 'cta'.
    * 'primary' is the default value.
    */
   variant?: ButtonVariant;
 }
 
-export type ButtonProps<T extends ElementType = "button"> = ButtonBaseProps<T> &
-  Omit<ComponentPropsWithoutRef<T>, keyof ButtonBaseProps<T>>;
-
-type PolymorphicButton = <T extends ElementType = "button">(
-  p: ButtonProps<T> & { ref?: polymorphicRef<T> }
-) => ReactElement<ButtonProps<T>>;
+export type ButtonProps = ButtonBaseProps &
```

Comment:
This could be simpler now and follow our usual patten of being in the interface?